### PR TITLE
Refactor : Enable CA2249

### PR DESCRIPTION
**Pull request by ShreyasJejurkar**
1. Enables CA2249
2. formatted `.editorconfig`

Contributes to https://github.com/dotnet/aspnetcore/issues/24055

### CODE REVIEW

**Comment by ShreyasJejurkar** [comment]
Maybe need to suppress the rule for this instance! Any thoughts!? @pranavkm 

https://github.com/dotnet/aspnetcore/blob/8e8a7c3bbee3ec17ea4593789929dc15284ef05a/src/Middleware/Rewrite/src/UrlActions/RedirectAction.cs#L55-L56

File: .editorconfig [hunk 1]
```diff
@@ -259,52 +262,79 @@ dotnet_diagnostic.IDE2000.severity = warning
 [{eng/tools/**.cs,**/{test,samples,perf}/**.cs}]
 # CA1018: Mark attributes with AttributeUsageAttribute
 dotnet_diagnostic.CA1018.severity = suggestion
+
```
Comment (pranavkm) [inline .editorconfig:265]: Could you undo this? The spaces were meant to make comments or description much clearer, but these are all suppressions for tests which don't need that.

File: .editorconfig [hunk 2]
```diff
@@ -77,6 +77,9 @@ charset = utf-8-bom
 
 [*.{cs,vb}]
 
+# CA2249: Use string.Contains instead of string.IndexOf to improve readability.
+dotnet_diagnostic.CA2249.severity = warning
```
Comment (pranavkm) [inline .editorconfig:81]: Can you order it so it appears around line 215 in this file?

File: src/Http/Http.Abstractions/src/HostString.cs [hunk 3]
```diff
@@ -172,7 +172,7 @@ public static HostString FromUriComponent(string uriComponent)
             {
                 // IPv6 without brackets ::1 is the only type of host with 2 or more colons
             }
-            else if (uriComponent.IndexOf("xn--", StringComparison.Ordinal) >= 0)
+            else if (uriComponent.Contains("xn--"))
```
Comment (pranavkm) [inline src/Http/Http.Abstractions/src/HostString.cs:175]:
```suggestion
            else if (uriComponent.Contains("xn--", StringComparison.Ordinal))
```

File: src/Middleware/Rewrite/src/RewriteRule.cs [hunk 4]
```diff
@@ -59,7 +59,7 @@ public virtual void ApplyRule(RewriteContext context)
                 result = "/";
             }
 
-            if (result.IndexOf(Uri.SchemeDelimiter, StringComparison.Ordinal) >= 0)
+            if (result.Contains(Uri.SchemeDelimiter))
```
Comment (pranavkm) [inline src/Middleware/Rewrite/src/RewriteRule.cs:62]:
```suggestion
            if (result.Contains(Uri.SchemeDelimiter, StringComparison.Ordinal))
```

File: src/Middleware/Rewrite/src/UrlActions/RewriteAction.cs [hunk 5]
```diff
@@ -59,7 +59,7 @@ public override void ApplyAction(RewriteContext context, BackReferenceCollection
         }
 
         // TODO PERF, substrings, object creation, etc.
-        if (pattern.IndexOf(Uri.SchemeDelimiter, StringComparison.Ordinal) >= 0)
+        if (pattern.Contains(Uri.SchemeDelimiter))
```
Comment (pranavkm) [inline src/Middleware/Rewrite/src/UrlActions/RewriteAction.cs:62]:
```suggestion
        if (pattern.Contains(Uri.SchemeDelimiter, StringComparison.Ordinal))
```

File: src/Mvc/Mvc.TagHelpers/src/TagHelperOutputExtensions.cs [hunk 6]
```diff
@@ -176,7 +176,7 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
 
         var encodedSpaceChars = SpaceChars.Where(x => !x.Equals('\u0020')).Select(x => htmlEncoder.Encode(x.ToString())).ToArray();
 
-        if (SpaceChars.Any(classValue.Contains) || encodedSpaceChars.Any(value => classValue.IndexOf(value, StringComparison.Ordinal) >= 0))
+        if (SpaceChars.Any(classValue.Contains) || encodedSpaceChars.Any(value => classValue.Contains(value)))
```
Comment (pranavkm) [inline src/Mvc/Mvc.TagHelpers/src/TagHelperOutputExtensions.cs:179]:
```suggestion
        if (SpaceChars.Any(classValue.Contains) || encodedSpaceChars.Any(value => classValue.Contains(value, StringComparison.Ordinal)))
```

File: src/Mvc/Mvc.TagHelpers/src/TagHelperOutputExtensions.cs [hunk 7]
```diff
@@ -233,7 +233,7 @@ public static void MergeAttributes(this TagHelperOutput tagHelperOutput, TagBuil
...
```
Comment (pranavkm) [inline src/Mvc/Mvc.TagHelpers/src/TagHelperOutputExtensions.cs:236]:
```suggestion
        if (SpaceChars.Any(classValue.Contains) || encodedSpaceChars.Any(value => classValue.Contains(value, StringComparison.Ordinal)))
```

**Review by pranavkm — CHANGES_REQUESTED** [review]

**Review by pranavkm — APPROVED** [review]

**Comment by ShreyasJejurkar** [comment]
Thanks, @pranavkm taking care of it. Sorry for the delay from my side. I guess we live in totally opposite timezone, I am in India! 

But anyway thanks for taking care and merging!

**Pull request merged**